irobotclient/request_handler.py

```diff
@@ -0,0 +1,161 @@
+                    raise IrobotClientException(errno=RESPONSES['DENIED_IRODS'].errno,
+                                                message="ERROR: Access to IRODs denied.")
+
+                elif response.status_code == RESPONSES['NOT_FOUND'].status_code:
+                    raise IrobotClientException(errno=RESPONSES['NOT_FOUND'].errno,
+                                                message="ERROR: The file requested cannot be found.  Please check the "
+                                                        "path and name of the requested file.")
+
+                elif response.status_code == RESPONSES['INVALID_REQUEST_METHOD'].status_code:
+                    raise IrobotClientException(errno=RESPONSES['INVALID_REQUEST_METHOD'].errno,
+                                                message="ERROR: Invalid HTTP request method; only GET, HEAD, POST, "
+                                                        "DELETE, and OPTIONS are supported.")
+
+                elif response.status_code == RESPONSES['INVALID_MEDIA_REQUESTED'].status_code:
+                    raise IrobotClientException(errno=RESPONSES['INVALID_MEDIA_REQUESTED'].errno,
+                                                message="ERROR: Unsupported HTTP media type requested.")
+
+                elif response.status_code == RESPONSES['INVALID_RANGE'].status_code:
+                    raise IrobotClientException(errno=RESPONSES['INVALID_RANGE'].errno,
+                                                message="ERROR: Invalid data range requested.")
+
+                elif response.status_code == RESPONSES['TIMEOUT'].status_code:
+                    raise IrobotClientException(errno=RESPONSES['TIMEOUT'].errno,
+                                                message="ERROR: Connection timeout from iRobot.")
+
+                elif response.status_code == RESPONSES['PRECACHE_FULL'].status_code:
+                    raise IrobotClientException(errno=RESPONSES['PRECACHE_FULL'].errno,
+                                                message="ERROR: Precache is full or too small for the size of the "
+                                                        "requested file.")
+
+                continue
+
+            else:
+                raise IrobotClientException(errno=errno.ECONNABORTED, message="ERROR: Maximum number of request "
+                                                                              "retries. Please try again later.")
+        except ConnectionError:
+            raise
+        except TimeoutError:
+            raise
+        except IrobotClientException:
+            raise
+        except:
+            raise
+
+    def _get_request_delay(self, response: requests.Response) -> int:
+        """
+
+        :return:
+        """
+
+        if "iRobot-ETA" in response.headers:
+            # Eg:  iRobot-ETA: 2017-09-25T12:34:56Z+00:00 +/- 123
+            stripped_response_eta = (response.headers["iRobot-ETA"].split('Z'))[0]
```

Comment:
I've updated this to reflect Chris' timezone offset error but other than that nothing has changed.  I've still had to split the +/- 123 off the end of the header string and use `strptime()` to create a datetime object.... So I haven't used any new library functions.